diff --git a/CMakeLists.txt b/CMakeLists.txt
index eeb0bb0b8f..f0461d5fab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -310,9 +310,9 @@ if (MAGMA_ENABLE_CUDA)
         )
     else()
         # No Position Independent Code on Windows. Compiler will complain if you add that flag.
-        target_compile_options(magma_nvcc_flags
+        target_compile_definitions(magma_nvcc_flags
             INTERFACE
-            $<$<COMPILE_LANGUAGE:CUDA>:--compiler-options;${FORTRAN_CONVENTION}>
+            $<$<COMPILE_LANGUAGE:CUDA>:${FORTRAN_CONVENTION}>
         )
     endif(COMPILER_SUPPORTS_FPIC)
 
diff --git a/control/magma_internal.h b/control/magma_internal.h
index 2310659235..8722c7fd3f 100644
--- a/control/magma_internal.h
+++ b/control/magma_internal.h
@@ -33,7 +33,7 @@
     // functions where Microsoft fails to provide C99 standard
     // (only with Microsoft, not with nvcc on Windows)
     // in both magma_internal.h and testings.h
-    #ifndef __NVCC__
+    #if !defined(__CUDACC__)
 
         #include <float.h>
         #define copysign(x,y) _copysign(x,y)
