diff --git a/cmake/HPX_GeneratePackageUtils.cmake b/cmake/HPX_GeneratePackageUtils.cmake
index c11c1e2..f2c057f 100644
--- a/cmake/HPX_GeneratePackageUtils.cmake	
+++ b/cmake/HPX_GeneratePackageUtils.cmake
@@ -232,6 +232,12 @@ function(hpx_sanitize_usage_requirements property is_build)
       string(REPLACE "$<INSTALL_INTERFACE:" "$<1:${CMAKE_INSTALL_PREFIX}/" prop
                      "${prop}"
       )
+      string(REPLACE "$<LINK_ONLY:" "$<1:" prop
+                     "${prop}"
+      )
+      string(REPLACE "$<TARGET_PDB_FILE::" "$<0:" prop
+                     "${prop}"
+      )
     endif()
     set(_sanitized_property ${_sanitized_property} ${prop})
   endforeach()
@@ -288,10 +291,12 @@ endfunction(hpx_construct_cflag_list)
 
 function(hpx_construct_library_list link_libraries link_options library_list)
   foreach(library IN LISTS ${link_libraries})
+    string(REPLACE "$<LINK_ONLY:" "$<1:" library "${library}")
+    string(REPLACE "$<TARGET_PDB_FILE:" "$<0:" library "${library}")
     if(IS_ABSOLUTE ${library})
       get_filename_component(_library_path ${library} DIRECTORY)
       get_filename_component(_library_name ${library} NAME_WE)
-      string(REPLACE ${CMAKE_SHARED_LIBRARY_PREFIX} "" _library_name
+      string(REPLACE "${CMAKE_SHARED_LIBRARY_PREFIX}" "" _library_name
                      ${_library_name}
       )
       set(_library_list
@@ -302,6 +306,8 @@ function(hpx_construct_library_list link_libraries link_options library_list)
     endif()
   endforeach()
   foreach(option IN LISTS ${link_options})
+    string(REPLACE "$<LINK_ONLY:" "$<1:" option "${option}")
+    string(REPLACE "$<TARGET_PDB_FILE:" "$<0:" option "${option}")
     set(_library_list "${_library_list} ${option}")
   endforeach()
   set(${library_list}
@@ -390,7 +393,8 @@ function(hpx_generate_pkgconfig_from_target target template is_build)
   )
 
   string(TOLOWER ${CMAKE_BUILD_TYPE} build_type)
-
+  message(STATUS "hpx_library_list")
+  message(STATUS "hpx_cflags_list")
   configure_file(
     cmake/templates/${template}.pc.in
     ${OUTPUT_DIR}${template}_${build_type}.pc.in @ONLY ESCAPE_QUOTES
diff --git a/cmake/templates/hpx_application.pc.in b/cmake/templates/hpx_application.pc.in
index f7b32d7..52a46d0 100644
--- a/cmake/templates/hpx_application.pc.in	
+++ b/cmake/templates/hpx_application.pc.in
@@ -5,9 +5,9 @@
 # SPDX-License-Identifier: BSL-1.0
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
-
+prefix=@CMAKE_INSTALL_PREFIX@
 Name: hpx_application
 Description: High Performance ParalleX (application configuration)$<$<CONFIG:Debug>: - debug build>
 Version: @HPX_VERSION@
 Libs: @hpx_library_list@
-Cflags: @CMAKE_CXX_FLAGS@ @hpx_cflags_list@
+Cflags: @hpx_cflags_list@
diff --git a/cmake/templates/hpx_component.pc.in b/cmake/templates/hpx_component.pc.in
index 38260fc..f4c72a2 100644
--- a/cmake/templates/hpx_component.pc.in	
+++ b/cmake/templates/hpx_component.pc.in
@@ -5,9 +5,9 @@
 # SPDX-License-Identifier: BSL-1.0
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
-
+prefix=@CMAKE_INSTALL_PREFIX@
 Name: hpx_component
 Description: High Performance ParalleX (component configuration)$<$<CONFIG:Debug>: - debug build>
 Version: @HPX_VERSION@
-Libs: -shared @hpx_library_list@
-Cflags: @CMAKE_CXX_FLAGS@ @hpx_cflags_list@
+Libs: @hpx_library_list@
+Cflags: @hpx_cflags_list@
diff --git a/libs/CMakeLists.txt b/libs/CMakeLists.txt
index 071918d..0feb772 100644
--- a/libs/CMakeLists.txt	
+++ b/libs/CMakeLists.txt
@@ -245,7 +245,7 @@ install(
 )
 
 # install PDB if needed
-if(MSVC)
+if(MSVC AND BUILD_SHARED_LIBS)
   install(
     FILES $<TARGET_PDB_FILE:hpx_full>
     DESTINATION ${CMAKE_INSTALL_BINDIR}
diff --git a/cmake/HPX_AddExecutable.cmake b/cmake/HPX_AddExecutable.cmake
index c2c35d4..114f786 100644
--- a/cmake/HPX_AddExecutable.cmake	
+++ b/cmake/HPX_AddExecutable.cmake
@@ -167,7 +167,7 @@ function(add_hpx_executable name)
     endif()
     set(_target_flags INSTALL INSTALL_FLAGS DESTINATION ${install_destination})
     # install PDB if needed
-    if(MSVC)
+    if(MSVC AND BUILD_SHARED_LIBS)
       # cmake-format: off
       set(_target_flags
           ${_target_flags}
diff --git a/cmake/HPX_AddComponent.cmake b/cmake/HPX_AddComponent.cmake
index c2c35d4..114f786 100644
--- a/cmake/HPX_AddComponent.cmake	
+++ b/cmake/HPX_AddComponent.cmake
@@ -201,7 +201,7 @@ function(add_hpx_component name)
     # cmake-format: on
 
     # install PDB if needed
-    if(MSVC)
+    if(MSVC AND BUILD_SHARED_LIBS)
       # cmake-format: off
       set(_target_flags
           ${_target_flags}
@@ -303,7 +303,7 @@ function(add_hpx_component name)
   hpx_setup_target(
     ${name}_component
     TYPE COMPONENT
-    NAME ${name}
+    NAME ${name}_component
     FOLDER ${${name}_FOLDER}
     COMPILE_FLAGS ${${name}_COMPILE_FLAGS}
     LINK_FLAGS ${${name}_LINK_FLAGS}
diff --git a/cmake/HPX_AddModule.cmake b/cmake/HPX_AddModule.cmake
index cc958bf..9fdd896 100644
--- a/cmake/HPX_AddModule.cmake	
+++ b/cmake/HPX_AddModule.cmake
@@ -363,7 +363,7 @@ function(add_hpx_module libname modulename)
     set_target_properties(hpx_${modulename} PROPERTIES UNITY_BUILD ON)
   endif()
 
-  if(MSVC)
+  if(MSVC AND BUILD_SHARED_LIBS)
     set_target_properties(
       hpx_${modulename}
       PROPERTIES COMPILE_PDB_NAME_DEBUG hpx_${modulename}d
