diff --git a/cmake/HPX_SetupAllocator.cmake b/cmake/HPX_SetupAllocator.cmake
index 221c988..f61d846 100644
--- a/cmake/HPX_SetupAllocator.cmake	
+++ b/cmake/HPX_SetupAllocator.cmake
@@ -95,7 +95,7 @@ if(NOT TARGET hpx_dependencies_allocator)
     # ##########################################################################
     # TBBMALLOC
     if("${HPX_WITH_MALLOC_UPPER}" STREQUAL "TBBMALLOC")
-      find_package(TBBmalloc)
+      find_package(TBB CONFIG REQUIRED)
-      if(NOT TBBMALLOC_LIBRARY AND NOT TBBMALLOC_PROXY_LIBRARY)
+      if(0)
         hpx_error(${allocator_error})
       endif()
@@ -105,8 +105,7 @@ if(NOT TARGET hpx_dependencies_allocator)
         )
       endif()
       target_link_libraries(
-        hpx_dependencies_allocator INTERFACE ${TBBMALLOC_LIBRARY}
-                                             ${TBBMALLOC_PROXY_LIBRARY}
+        hpx_dependencies_allocator INTERFACE TBB::tbbmalloc
       )
     endif()
 
