diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ad5f7a..6084949 100644
--- a/CMakeLists.txt	
+++ b/CMakeLists.txt
@@ -222,6 +222,9 @@ else()
             DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}
             COMPONENT devel)
     file(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}Config.cmake
+               "include(CMakeFindDependencyMacro)\n"
+               "find_dependency(PkgConfig)\n" 
+               "pkg_search_module(HWLOC hwloc IMPORTED_TARGET)\n" 
                "include(\${CMAKE_CURRENT_LIST_DIR}/${PROJECT_NAME}Targets.cmake)\n")
 
     write_basic_package_version_file("${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}ConfigVersion.cmake"
