diff --git a/here-src-6-7c89855f8c.clean/src/corelib/kernel/qproperty.h b/here-src-6-7c89855f8c/src/corelib/kernel/qproperty.h
index 610a390..339cae2 100644
--- a/here-src-6-7c89855f8c.clean/src/corelib/kernel/qproperty.h
+++ b/here-src-6-7c89855f8c/src/corelib/kernel/qproperty.h
@@ -18,6 +18,7 @@
 #if defined(__cpp_lib_source_location)
 #define QT_SOURCE_LOCATION_NAMESPACE std
 #define QT_PROPERTY_COLLECT_BINDING_LOCATION
+#define QT_HAS_SOURCE_LOCATION
 #if defined(Q_CC_MSVC)
 /* MSVC runs into an issue with constexpr with source location (error C7595)
    so use the factory function as a workaround */
@@ -98,7 +99,7 @@ struct Q_CORE_EXPORT QPropertyBindingSourceLocation
     quint32 line = 0;
     quint32 column = 0;
     QPropertyBindingSourceLocation() = default;
-#ifdef __cpp_lib_source_location
+#ifdef QT_HAS_SOURCE_LOCATION
     constexpr QPropertyBindingSourceLocation(const std::source_location &cppLocation)
     {
         fileName = cppLocation.file_name();
