From d49f5752a560d7c8fec99c57bbfb192a9be1b296 Mon Sep 17 00:00:00 2001
From: Alexander Neumann <Alexander.Neumann@hamburg.de>
Date: Sun, 20 Feb 2022 10:12:43 +0100
Subject: [PATCH] Fix QMetaType::typeName returning wrong result for clang-cl

Pick-to: 6.2 6.3
Change-Id: I94fa1d56d98b8caae514b37fbf6432985195d827
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
---

diff --git a/src/corelib/kernel/qmetatype.h b/src/corelib/kernel/qmetatype.h
index 4b13db9..615ea52 100644
--- a/src/corelib/kernel/qmetatype.h
+++ b/src/corelib/kernel/qmetatype.h
@@ -2185,7 +2185,9 @@
 #ifdef QT_NAMESPACE
             QT_STRINGIFY(QT_NAMESPACE) "::"
 #endif
-#ifdef Q_CC_MSVC
+#if defined(Q_CC_MSVC) && defined(Q_CC_CLANG)
+            "auto __cdecl QtPrivate::typenameHelper(void) [T = "
+#elif defined(Q_CC_MSVC)
             "auto __cdecl QtPrivate::typenameHelper<"
 #elif defined(Q_CC_CLANG)
             "auto QtPrivate::typenameHelper() [T = "
@@ -2195,7 +2197,7 @@
             "constexpr auto QtPrivate::typenameHelper() [with T = "
 #endif
             ) - 1;
-#ifdef Q_CC_MSVC
+#if defined(Q_CC_MSVC) && !defined(Q_CC_CLANG)
         constexpr int suffix = sizeof(">(void)");
 #else
         constexpr int suffix = sizeof("]");
