diff --git a/src/corelib/global/qendian.h b/src/corelib/global/qendian.h
index 9cd0044cb..72cc78e7f 100644
--- a/src/corelib/global/qendian.h	
+++ b/src/corelib/global/qendian.h
@@ -36,11 +36,7 @@ template <typename T> Q_ALWAYS_INLINE void qToUnaligned(const T src, void *dest)
     // Using sizeof(T) inside memcpy function produces internal compiler error with
     // MSVC2008/ARM in tst_endian -> use extra indirection to resolve size of T.
     const size_t size = sizeof(T);
-#if __has_builtin(__builtin_memcpy)
-    __builtin_memcpy
-#else
     memcpy
-#endif
             (dest, &src, size);
 }
 
@@ -48,11 +44,7 @@ template <typename T> Q_ALWAYS_INLINE T qFromUnaligned(const void *src)
 {
     T dest;
     const size_t size = sizeof(T);
-#if __has_builtin(__builtin_memcpy)
-    __builtin_memcpy
-#else
     memcpy
-#endif
             (&dest, src, size);
     return dest;
 }
