diff --git a/omniORB-4-5a27911dc0.clean/config/config.mk b/omniORB-4-5a27911dc0/config/config.mk
index 31d2cfd..c8e34f5 100644
--- a/omniORB-4-5a27911dc0.clean/config/config.mk
+++ b/omniORB-4-5a27911dc0/config/config.mk
@@ -21,7 +21,7 @@
 # You should also look at <top>/mk/platforms/$(platform).mk and if necessary
 # edit the make variables, such as CC and CXX, in the file.
 
-#platform = x86_win32_vs_16
+platform = x86_win32_vs_16
 #platform = x86_win32_vs_15
 #platform = x86_win32_vs_14
 #platform = x86_win32_vs_12
diff --git a/omniORB-4-5a27911dc0.clean/configure b/omniORB-4-5a27911dc0/configure
index a90b79e..360d7b0 100644
--- a/omniORB-4-5a27911dc0.clean/configure
+++ b/omniORB-4-5a27911dc0/configure
@@ -8435,6 +8435,7 @@ case "$host" in
   *-*-openbsd*)  plat_name="OpenBSD";  plat_def="__openbsd__";  os_v="3";;
   *-*-sco*)      plat_name="OSR5";     plat_def="__osr5__";     os_v="5";;
   *-*-gnu*)      plat_name="GNU";      plat_def="__hurd__" ;    os_v="0";;
+  *-*-mingw*)    plat_name="Win32Platform";      plat_def="__WIN32__" ;    os_v="4";;
 esac
 
 PLATFORM_NAME=$plat_name
diff --git a/omniORB-4-5a27911dc0.clean/configure.ac b/omniORB-4-5a27911dc0/configure.ac
index d485463..2ba9c82 100644
--- a/omniORB-4-5a27911dc0.clean/configure.ac
+++ b/omniORB-4-5a27911dc0/configure.ac
@@ -231,6 +231,7 @@ case "$host" in
   *-*-openbsd*)  plat_name="OpenBSD";  plat_def="__openbsd__";  os_v="3";;
   *-*-sco*)      plat_name="OSR5";     plat_def="__osr5__";     os_v="5";;
   *-*-gnu*)      plat_name="GNU";      plat_def="__hurd__" ;    os_v="0";;
+  *-*-mingw*)    plat_name="Win32Platform";      plat_def="__WIN32__" ;    os_v="4";;
 esac
 
 AC_SUBST(PLATFORM_NAME, $plat_name)
diff --git a/omniORB-4-5a27911dc0.clean/include/omnithread/atomic.h b/omniORB-4-5a27911dc0/include/omnithread/atomic.h
index 3fed8df..e1b6a71 100644
--- a/omniORB-4-5a27911dc0.clean/include/omnithread/atomic.h
+++ b/omniORB-4-5a27911dc0/include/omnithread/atomic.h
@@ -27,7 +27,7 @@
 #define __omnithread_atomic_h_
 
 
-#ifdef OMNI_HAVE_SYNC_ADD_AND_FETCH
+#if defined(OMNI_HAVE_SYNC_ADD_AND_FETCH) && !defined(_MSC_VER)
 
 #  define OMNI_REFCOUNT_DEFINED
 
diff --git a/omniORB-4-5a27911dc0.clean/mk/beforeauto.mk.in b/omniORB-4-5a27911dc0/mk/beforeauto.mk.in
index f078912..9aef646 100644
--- a/omniORB-4-5a27911dc0.clean/mk/beforeauto.mk.in
+++ b/omniORB-4-5a27911dc0/mk/beforeauto.mk.in
@@ -4,7 +4,7 @@
 #
 # Based upon beforedir.mk and platform makefiles
 
-platform         = autoconf
+# platform         = autoconf
 MAKEFILE_INC_DIR = $(BASE_OMNI_TREE)/$(CURRENT)/
 IMPORT_TREES     = $(TOP) $(BASE_OMNI_TREE)
 THIS_IMPORT_TREE = $(TOP)
@@ -341,8 +341,8 @@ GENERATED_CXX_HDRS += $(CORBA_STUB_HDRS)
 # later if necessary.
 #
 
-UnixPlatform = 1
-ThreadSystem = Posix
+# UnixPlatform = 1
+# ThreadSystem = Posix
 
 #
 # General rules for cleaning.
@@ -601,14 +601,17 @@ OMNIORB_COS_DYNAMIC_LIB_DEPEND := $(GENERATE_LIB_DEPEND)
 # the rules in their platform files.
 #
 BuildSharedLibrary = 1       # Enable by default
-SHAREDLIB_SUFFIX = so
+SHAREDLIB_SUFFIX = dll
 
-SharedLibraryFullNameTemplate 	= lib$$1$$2.$(SHAREDLIB_SUFFIX).$$3.$$4
-SharedLibrarySoNameTemplate   	= lib$$1$$2.$(SHAREDLIB_SUFFIX).$$3
-SharedLibraryLibNameTemplate  	= lib$$1$$2.$(SHAREDLIB_SUFFIX)
-SharedLibraryImplibNameTemplate = lib$$1$$2.a
+# .$$3.$$4
+# .$$3
 
-SharedLibraryPlatformLinkFlagsTemplate = -shared -Wl,-soname,$$soname
+SharedLibraryFullNameTemplate 	= $$1$$2.$(SHAREDLIB_SUFFIX)
+SharedLibrarySoNameTemplate   	= $$1$$2.$(SHAREDLIB_SUFFIX)
+SharedLibraryLibNameTemplate  	= $$1$$2.$(SHAREDLIB_SUFFIX)
+SharedLibraryImplibNameTemplate = $$1$$2.lib
+
+# SharedLibraryPlatformLinkFlagsTemplate = -shared -LD$${debug:+d} -Wl,-soname,$$soname
 
 define SharedLibraryFullName
 fn() { \
@@ -655,13 +658,13 @@ define ExportSharedLibraryToDir
  $(ExportLibraryToDir); \
  $(ParseNameSpec); \
  soname=$(SharedLibrarySoNameTemplate); \
- libname=$(SharedLibraryLibNameTemplate); \
- set -x; \
- cd $$dir; \
- $(RM) $$soname; \
- ln -s $(<F) $$soname; \
- $(RM) $$libname; \
- ln -s $$soname $$libname;
+ libname=$(SharedLibraryLibNameTemplate);
+ # set -x; \
+ # cd $$dir; \
+ # $(RM) $$soname; \
+ # ln -s $(<F) $$soname; \
+ # $(RM) $$libname; \
+ # ln -s $$soname $$libname;
 endef
 
 define ExportSharedLibrary
@@ -738,7 +741,7 @@ shared/%.o: %.c
 # these such as AR, RM, etc (see section 10.3 of the GNU make manual).
 #
 
-AR              = ar cq
+AR              = @AR@ cq
 CC              = @CC@
 CXX             = @CXX@
 RANLIB		= @RANLIB@
@@ -1311,3 +1314,6 @@ endif
 
 lib_depend := $(patsubst %,$(LibPattern),omnithread)
 OMNITHREAD_LIB_DEPEND := $(GENERATE_LIB_DEPEND)
+
+platform = x86_win32_vs_16
+include $(THIS_IMPORT_TREE)/mk/platforms/$(platform).mk
\ No newline at end of file
diff --git a/omniORB-4-5a27911dc0.clean/mk/beforedir.mk b/omniORB-4-5a27911dc0/mk/beforedir.mk
index 855bc4d..e1c4ef7 100644
--- a/omniORB-4-5a27911dc0.clean/mk/beforedir.mk
+++ b/omniORB-4-5a27911dc0/mk/beforedir.mk
@@ -377,6 +377,7 @@ export::
 
 else
 
+THIS_IMPORT_TREE := $(TOP)
 include $(THIS_IMPORT_TREE)/mk/platforms/$(platform).mk
 
 endif
diff --git a/omniORB-4-5a27911dc0.clean/mk/mklib.mk b/omniORB-4-5a27911dc0/mk/mklib.mk
index fb83cf3..117a365 100644
--- a/omniORB-4-5a27911dc0.clean/mk/mklib.mk
+++ b/omniORB-4-5a27911dc0/mk/mklib.mk
@@ -16,12 +16,26 @@
 # are called libfoo.a and libfoo.so.1.3 etc.
 #
 
-all:: mkstatic mkshared
+ifdef BuildSharedLibrary
+
+all:: mkshared
+
+export:: mkshared
+
+ifdef INSTALLTARGET
+install:: mkshared
+endif
+
+else
+
+all:: mkstatic
 
-export:: mkstatic mkshared
+export:: mkstatic
 
 ifdef INSTALLTARGET
-install:: mkstatic mkshared
+install:: mkstatic
+endif
+
 endif
 
 vers := $(subst ., ,$(LIB_VERSION))
@@ -38,7 +52,7 @@ namespec := $(LIB_NAME) $(vers)
 # Build Static library
 ##############################################################################
 
-ifndef NoStaticLibrary
+ifndef BuildSharedLibrary
 
 staticlib := static/$(patsubst %,$(LibNoDebugPattern),$(LIB_NAME)$(major))
 MDFLAGS += -p static/
@@ -46,7 +60,7 @@ MDFLAGS += -p static/
 mkstatic::
 	@(dir=static; $(CreateDir))
 
-mkstatic:: $(staticlib)
+$(warning mkstatic) mkstatic:: $(staticlib)
 
 $(staticlib): $(patsubst %, static/%, $(LIB_OBJS))
 	@$(StaticLinkLibrary)
@@ -71,7 +85,7 @@ veryclean::
 
 else
 
-mkstatic::
+# $(warning mkstatic dummy) mkstatic::
 
 endif
 
@@ -97,10 +111,10 @@ mkshared::
 mkshared:: $(shlib)
 
 $(shlib): $(patsubst %, shared/%, $(LIB_OBJS) $(LIB_SHARED_ONLY_OBJS))
-	@(namespec="$(namespec)" extralibs="$(imps) $(extralibs)"; \
-         $(MakeCXXSharedLibrary))
+	$(warning shlib test) @(namespec="$(namespec)" extralibs="$(imps) $(extralibs)"; \
+         $(warning mksharedline)$(MakeCXXSharedLibrary))
 
-export:: $(shlib)
+ $(warning export mklib) export:: $(shlib)
 	@(namespec="$(namespec)"; \
           $(ExportSharedLibrary))
 
@@ -139,7 +153,7 @@ endif
 
 else
 
-mkshared::
+# mkshared::
 
 endif
 
@@ -150,9 +164,9 @@ ifdef Win32Platform
 
 ifdef BuildSharedLibrary
 
-all:: mkstaticdbug mkshareddbug
+all:: mkshareddbug
 
-export:: mkstaticdbug mkshareddbug
+export:: mkshareddbug
 
 else
 
@@ -166,6 +180,8 @@ endif
 #      Static debug libraries
 #####################################################
 
+ifndef BuildSharedLibrary
+
 dbuglib := debug/$(patsubst %,$(LibDebugPattern),$(LIB_NAME)$(major))
 MDFLAGS += -p debug/
 
@@ -187,7 +203,8 @@ clean::
 veryclean::
 	$(RM) debug/*.o
 	$(RM) $(dbuglib)
-
+  
+endif
 #####################################################
 #      DLL debug libraries
 #####################################################
diff --git a/omniORB-4-5a27911dc0.clean/mk/platforms/x86_win32_vs_16.mk b/omniORB-4-5a27911dc0/mk/platforms/x86_win32_vs_16.mk
index cd2f16a..44370fb 100644
--- a/omniORB-4-5a27911dc0.clean/mk/platforms/x86_win32_vs_16.mk
+++ b/omniORB-4-5a27911dc0/mk/platforms/x86_win32_vs_16.mk
@@ -8,7 +8,7 @@ x86Processor = 1
 
 compiler_version_suffix=_vc16
 
-WINVER = 0x0501
+WINVER = 0x0A00
 
 BINDIR = bin/x86_win32
 LIBDIR = lib/x86_win32
@@ -27,28 +27,28 @@ ABSTOP = $(shell cd $(TOP); pwd)
 
 # Use the following set of flags to build and use multithreaded DLLs
 #
-MSVC_DLL_CXXNODEBUGFLAGS       = -MD -EHs -GS -GR -Zi -nologo
+MSVC_DLL_CXXNODEBUGFLAGS       = -MD -EHs -GS -GR -Z7 -nologo
 MSVC_DLL_CXXLINKNODEBUGOPTIONS = -nologo -manifest -DEBUG
-MSVC_DLL_CNODEBUGFLAGS         = -MD -GS -GR -Zi -nologo
+MSVC_DLL_CNODEBUGFLAGS         = -MD -GS -GR -Z7 -nologo
 MSVC_DLL_CLINKNODEBUGOPTIONS   = -nologo -manifest -DEBUG
 #
-MSVC_DLL_CXXDEBUGFLAGS         = -MDd -EHs -RTC1 -GS -GR -Zi -nologo
+MSVC_DLL_CXXDEBUGFLAGS         = -MDd -EHs -RTC1 -GS -GR -Z7 -nologo
 MSVC_DLL_CXXLINKDEBUGOPTIONS   = -nologo -manifest -DEBUG
-MSVC_DLL_CDEBUGFLAGS           = -MDd -RTC1 -GS -GR -Zd -Zi -nologo
+MSVC_DLL_CDEBUGFLAGS           = -MDd -RTC1 -GS -GR -Zd -Z7 -nologo
 MSVC_DLL_CLINKDEBUGOPTIONS     = -nologo -manifest -DEBUG
 #
 # Or
 #
 # Use the following set of flags to build and use multithread static libraries
 #
-MSVC_STATICLIB_CXXNODEBUGFLAGS       = -MT -EHs -GS -GR -Zi -nologo
+MSVC_STATICLIB_CXXNODEBUGFLAGS       = -MT -EHs -GS -GR -Z7 -nologo
 MSVC_STATICLIB_CXXLINKNODEBUGOPTIONS = -nologo -manifest -DEBUG
-MSVC_STATICLIB_CNODEBUGFLAGS         = -MT -GS -GR -Zi -nologo
+MSVC_STATICLIB_CNODEBUGFLAGS         = -MT -GS -GR -Z7 -nologo
 MSVC_STATICLIB_CLINKNODEBUGOPTIONS   = -nologo -manifest -DEBUG
 
-MSVC_STATICLIB_CXXDEBUGFLAGS         = -MTd -EHs -RTC1 -GS -GR -Zi -nologo
+MSVC_STATICLIB_CXXDEBUGFLAGS         = -MTd -EHs -RTC1 -GS -GR -Z7 -nologo
 MSVC_STATICLIB_CXXLINKDEBUGOPTIONS   = -nologo -manifest -DEBUG
-MSVC_STATICLIB_CDEBUGFLAGS           = -MTd -RTC1 -GS -GR -Zi -nologo
+MSVC_STATICLIB_CDEBUGFLAGS           = -MTd -RTC1 -GS -GR -Z7 -nologo
 MSVC_STATICLIB_CLINKDEBUGOPTIONS     = -nologo -manifest -DEBUG
 
 
@@ -65,7 +65,7 @@ IMPORT_CPPFLAGS += -D__x86__ -D__NT__ -D__OSVERSION__=4 \
 
 
 # Default directory for the omniNames log files.
-OMNINAMES_LOG_DEFAULT_LOCATION = C:\\temp
+#OMNINAMES_LOG_DEFAULT_LOCATION = temp
 
 
 # Add the location of the Open SSL library
diff --git a/omniORB-4-5a27911dc0.clean/mk/python.mk b/omniORB-4-5a27911dc0/mk/python.mk
index 1380018..fb321a5 100644
--- a/omniORB-4-5a27911dc0.clean/mk/python.mk
+++ b/omniORB-4-5a27911dc0/mk/python.mk
@@ -4,7 +4,7 @@
 
 PYVERSION := $(shell $(PYTHON) -c 'import sys; sys.stdout.write(".".join(sys.version.split(".")[:2]))')
 PYPREFIX  := $(shell $(PYTHON) -c 'import sys; sys.stdout.write(sys.exec_prefix.replace("\\","/"))')
-PYINCDIR  := $(shell $(PYTHON) -c 'import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_python_inc().replace("\\","/"))')
+PYINCDIR  := $(prefix)/include/python3.10
 
 PythonSHAREDLIB_SUFFIX = $(shell $(PYTHON) -c 'import sys, distutils.sysconfig; sys.stdout.write((distutils.sysconfig.get_config_var("SO") or ".so").lstrip("."))')
 
diff --git a/omniORB-4-5a27911dc0.clean/mk/unix.mk b/omniORB-4-5a27911dc0/mk/unix.mk
index 58d83a2..778ea86 100644
--- a/omniORB-4-5a27911dc0.clean/mk/unix.mk
+++ b/omniORB-4-5a27911dc0/mk/unix.mk
@@ -3,7 +3,7 @@
 #
 
 UnixPlatform = 1
-
+agagasf asfaedf 
 #
 # Any recursively-expanded variable set here can be overridden _afterwards_ by
 # a platform-specific mk file which includes this one.
diff --git a/omniORB-4-5a27911dc0.clean/mk/win32.mk b/omniORB-4-5a27911dc0/mk/win32.mk
index 94c9aad..f788d41 100644
--- a/omniORB-4-5a27911dc0.clean/mk/win32.mk
+++ b/omniORB-4-5a27911dc0/mk/win32.mk
@@ -81,7 +81,7 @@ endif
 
 
 ifndef WINVER
-WINVER = 0x0400
+WINVER = 0x0A00
 endif
 
 IMPORT_CPPFLAGS += -D__WIN32__ -D_WIN32_WINNT=$(WINVER)
@@ -179,7 +179,7 @@ define CXXExecutable
  $(RM) $@; \
  $(CXXLINK) -out:$@ $(CXXLINKOPTIONS) -PDB:$@.pdb $(IMPORT_LIBRARY_FLAGS) \
       $(filter-out $(LibPattern),$^) $$libs; \
- $(MANIFESTTOOL) /outputresource:"$@;#1" /manifest $@.manifest; \
+ $(MANIFESTTOOL) -outputresource:"$@;#1" -manifest $@.manifest; \
 )
 endef
 
@@ -187,7 +187,7 @@ define CExecutable
 (set -x; \
  $(RM) $@; \
  $(CLINK) -out:$@ $(CLINKOPTIONS) -PDB:$@.pdb $(IMPORT_LIBRARY_FLAGS) $(filter-out $(LibPattern),$^) $$libs; \
- $(MANIFESTTOOL) /outputresource:"$@;#1" /manifest $@.manifest; \
+ $(MANIFESTTOOL) -outputresource:"$@;#1" -manifest $@.manifest; \
 )
 endef
 
@@ -211,8 +211,8 @@ ifndef EmbeddedSystem
 BuildSharedLibrary = 1
 endif
 
-SharedLibraryFullNameTemplate = $(SharedLibraryLibNameTemplate).lib
 SharedLibraryLibNameTemplate  = $$1$$2$$3$$4_rt$${extrasuffix:-}
+SharedLibraryFullNameTemplate = $(SharedLibraryLibNameTemplate).lib
 SharedLibraryShortLibName = $$1$$2_rt$${extrasuffix:-}.lib
 SharedLibraryDllNameTemplate  = $$1$$2$$3$$4$(compiler_version_suffix)_rt$${extrasuffix:-}
 SharedLibraryExportSymbolFileNameTemplate = $$1$$2$${extrasuffix:-}.def
@@ -356,18 +356,21 @@ extralinkoption="$(MSVC_DLL_CXXLINKDEBUGOPTIONS)"; \
 else \
 extralinkoption="$(MSVC_DLL_CXXLINKNODEBUGOPTIONS)"; \
 fi; \
+$(warning Test MAKECXXSHARED) \
 if [ -z "$$nodeffile" ]; then \
 $(MakeCXXExportSymbolDefinitionFile) \
 defflag="-def:$$defname"; \
 fi; \
 $(MakeResourceDefinitionFile) \
+$(warning Test MAKECXXSHARED2) \
 set -x; \
 $(RM) $@; \
 $(CXXLINK) -out:$$dllname -DLL $$extralinkoption \
 $$defflag -IMPLIB:$@ $(IMPORT_LIBRARY_FLAGS) \
 $^ $$extralibs $$resname; \
-$(MANIFESTTOOL) /outputresource:"$$dllname;#2" /manifest $$dllname.manifest; \
-$(CP) $@ $$slibname;
+$(MANIFESTTOOL) -outputresource:"$$dllname;#2" -manifest $$dllname.manifest; \
+$(CP) $@ $$slibname; \
+$(warning Test MAKECXXSHARED3)
 endef
 
 # Export SharedLibrary
diff --git a/omniORB-4-5a27911dc0.clean/src/lib/omniORB/orbcore/dir.mk b/omniORB-4-5a27911dc0/src/lib/omniORB/orbcore/dir.mk
index 634e6b1..5c9558f 100644
--- a/omniORB-4-5a27911dc0.clean/src/lib/omniORB/orbcore/dir.mk
+++ b/omniORB-4-5a27911dc0/src/lib/omniORB/orbcore/dir.mk
@@ -220,7 +220,7 @@ include $(BASE_OMNI_TREE)/mk/mklib.mk
 
 #########################################################################
 ifdef Win32Platform
-
+ifndef BuildSharedLibrary
 stublib = static/$(patsubst %,$(LibNoDebugPattern),msvcstub)
 
 all:: $(stublib)
@@ -252,7 +252,7 @@ clean::
 
 veryclean::
 	$(RM) $(stubdblib)
-
+endif
 endif
 
 #########################################################################
diff --git a/omniORB-4-5a27911dc0.clean/src/lib/omnithread/dir.mk b/omniORB-4-5a27911dc0/src/lib/omnithread/dir.mk
index ed2194c..efaacfe 100644
--- a/omniORB-4-5a27911dc0.clean/src/lib/omnithread/dir.mk
+++ b/omniORB-4-5a27911dc0/src/lib/omnithread/dir.mk
@@ -32,14 +32,27 @@ LIB_VERSION  := $(OMNITHREAD_VERSION)
 LIB_OBJS     := $(CXXSRCS:.cc=.o)
 LIB_IMPORTS  := $(OMNITHREAD_PLATFORM_LIB)
 
-all:: mkstatic mkshared
+ifdef BuildSharedLibrary
+
+all:: mkshared
 
-export:: mkstatic mkshared
+export:: mkshared
 
 ifdef INSTALLTARGET
-install:: mkstatic mkshared
+install:: mkshared
 endif
 
+else
+
+all:: mkstatic
+
+export:: mkstatic
+
+ifdef INSTALLTARGET
+install:: mkstatic
+endif
+
+endif
 vers := $(subst ., ,$(LIB_VERSION))
 ifeq ($(words $(vers)), 2)
   vers  := _ $(vers)
@@ -54,7 +67,7 @@ namespec := $(LIB_NAME) $(vers)
 # Build Static library
 ##############################################################################
 
-ifndef NoStaticLibrary
+ifndef BuildSharedLibrary
 
 staticlib := static/$(patsubst %,$(LibNoDebugPattern),$(LIB_NAME)$(major))
 
@@ -84,10 +97,6 @@ veryclean::
 	$(RM) $(staticlib)
 	(dir=static; $(CleanStaticLibrary))
 
-else
-
-mkstatic::
-
 endif
 
 
@@ -152,10 +161,6 @@ veryclean::
 
 endif
 
-else
-
-mkshared::
-
 endif
 
 ##############################################################################
@@ -165,9 +170,9 @@ ifdef Win32Platform
 
 ifdef BuildSharedLibrary
 
-all:: mkstaticdbug mkshareddbug
+all:: mkshareddbug
 
-export:: mkstaticdbug mkshareddbug
+export:: mkshareddbug
 
 else
 
@@ -182,6 +187,8 @@ endif
 #      Static debug libraries
 #####################################################
 
+ifndef BuildSharedLibrary
+
 dbuglib := debug/$(patsubst %,$(LibDebugPattern),$(LIB_NAME)$(major))
 
 mkstaticdbug::
@@ -205,6 +212,8 @@ veryclean::
 	$(RM) $(dbuglib)
 	(dir=debug; $(CleanStaticLibrary))
 
+endif
+
 #####################################################
 #      DLL debug libraries
 #####################################################
diff --git a/omniORB-4-5a27911dc0.clean/src/tool/omniidl/cxx/dir.mk b/omniORB-4-5a27911dc0/src/tool/omniidl/cxx/dir.mk
index 84f10ac..fee51c7 100644
--- a/omniORB-4-5a27911dc0.clean/src/tool/omniidl/cxx/dir.mk
+++ b/omniORB-4-5a27911dc0/src/tool/omniidl/cxx/dir.mk
@@ -173,7 +173,7 @@ veryclean::
 
 ifdef Cygwin
 
-SharedLibraryPlatformLinkFlagsTemplate = -shared -Wl,-soname=$$soname,--export-dynamic,--enable-auto-import
+#SharedLibraryPlatformLinkFlagsTemplate = -shared -Wl,-soname=$$soname,--export-dynamic,--enable-auto-import
 
 extralibs += -L$(PYPREFIX)/lib/python$(PYVERSION)/config \
  -lpython$(PYVERSION).dll
