diff --git a/config/config.mk b/config/config.mk
index 31d2cfd..c8e34f5 100644
--- a/config/config.mk
+++ b/config/config.mk
@@ -21,7 +21,7 @@
 # You should also look at <top>/mk/platforms/$(platform).mk and if necessary
 # edit the make variables, such as CC and CXX, in the file.
 
-#platform = x86_win32_vs_16
+platform = x86_win32_vs_16
 #platform = x86_win32_vs_15
 #platform = x86_win32_vs_14
 #platform = x86_win32_vs_12
diff --git a/configure.ac b/configure.ac
index d485463..2ba9c82 100644
--- a/configure.ac
+++ b/configure.ac
@@ -231,6 +231,7 @@ case "$host" in
   *-*-openbsd*)  plat_name="OpenBSD";  plat_def="__openbsd__";  os_v="3";;
   *-*-sco*)      plat_name="OSR5";     plat_def="__osr5__";     os_v="5";;
   *-*-gnu*)      plat_name="GNU";      plat_def="__hurd__" ;    os_v="0";;
+  *-*-mingw*)    plat_name="Win32Platform";      plat_def="__WIN32__" ;    os_v="4";;
 esac
 
 AC_SUBST(PLATFORM_NAME, $plat_name)
diff --git a/include/omnithread/atomic.h b/include/omnithread/atomic.h
index 3fed8df..e1b6a71 100644
--- a/include/omnithread/atomic.h
+++ b/include/omnithread/atomic.h
@@ -27,7 +27,7 @@
 #define __omnithread_atomic_h_
 
 
-#ifdef OMNI_HAVE_SYNC_ADD_AND_FETCH
+#if defined(OMNI_HAVE_SYNC_ADD_AND_FETCH) && !defined(_MSC_VER)
 
 #  define OMNI_REFCOUNT_DEFINED
 
diff --git a/mk/beforeauto.mk.in b/mk/beforeauto.mk.in
index f078912..9aef646 100644
--- a/mk/beforeauto.mk.in
+++ b/mk/beforeauto.mk.in
@@ -738,7 +741,7 @@ shared/%.o: %.c
 # these such as AR, RM, etc (see section 10.3 of the GNU make manual).
 #
 
-AR              = ar cq
+AR              = @AR@ cq
 CC              = @CC@
 CXX             = @CXX@
 RANLIB		= @RANLIB@
@@ -1311,3 +1314,7 @@ endif
 
 lib_depend := $(patsubst %,$(LibPattern),omnithread)
 OMNITHREAD_LIB_DEPEND := $(GENERATE_LIB_DEPEND)
+
+ifdef Win32Platform
+include $(THIS_IMPORT_TREE)/mk/platforms/vcpkg.mk
+endif
\ No newline at end of file
diff --git a/mk/python.mk b/mk/python.mk
index 1380018..fb321a5 100644
--- a/mk/python.mk
+++ b/mk/python.mk
@@ -4,7 +4,7 @@
 
 PYVERSION := $(shell $(PYTHON) -c 'import sys; sys.stdout.write(".".join(sys.version.split(".")[:2]))')
 PYPREFIX  := $(shell $(PYTHON) -c 'import sys; sys.stdout.write(sys.exec_prefix.replace("\\","/"))')
-PYINCDIR  := $(shell $(PYTHON) -c 'import sys, distutils.sysconfig; sys.stdout.write(distutils.sysconfig.get_python_inc().replace("\\","/"))')
+PYINCDIR  := $(prefix)/include/python3.10
 
 PythonSHAREDLIB_SUFFIX = $(shell $(PYTHON) -c 'import sys, distutils.sysconfig; sys.stdout.write((distutils.sysconfig.get_config_var("SO") or ".so").lstrip("."))')
 
diff --git a/mk/win32.mk b/mk/win32.mk
index 94c9aad..f788d41 100644
--- a/mk/win32.mk
+++ b/mk/win32.mk
@@ -81,7 +81,6 @@ endif
 
 
 ifndef WINVER
-WINVER = 0x0400
 endif
 
 IMPORT_CPPFLAGS += -D__WIN32__ -D_WIN32_WINNT=$(WINVER)
@@ -179,7 +179,7 @@ define CXXExecutable
  $(RM) $@; \
  $(CXXLINK) -out:$@ $(CXXLINKOPTIONS) -PDB:$@.pdb $(IMPORT_LIBRARY_FLAGS) \
       $(filter-out $(LibPattern),$^) $$libs; \
- $(MANIFESTTOOL) /outputresource:"$@;#1" /manifest $@.manifest; \
+ $(MANIFESTTOOL) -outputresource:"$@;#1" -manifest $@.manifest; \
 )
 endef
 
@@ -187,7 +187,7 @@ define CExecutable
 (set -x; \
  $(RM) $@; \
  $(CLINK) -out:$@ $(CLINKOPTIONS) -PDB:$@.pdb $(IMPORT_LIBRARY_FLAGS) $(filter-out $(LibPattern),$^) $$libs; \
- $(MANIFESTTOOL) /outputresource:"$@;#1" /manifest $@.manifest; \
+ $(MANIFESTTOOL) -outputresource:"$@;#1" -manifest $@.manifest; \
 )
 endef
 
@@ -356,18 +356,18 @@ extralinkoption="$(MSVC_DLL_CXXLINKDEBUGOPTIONS)"; \
 else \
 extralinkoption="$(MSVC_DLL_CXXLINKNODEBUGOPTIONS)"; \
 fi; \
 if [ -z "$$nodeffile" ]; then \
 $(MakeCXXExportSymbolDefinitionFile) \
 defflag="-def:$$defname"; \
 fi; \
 $(MakeResourceDefinitionFile) \
 set -x; \
 $(RM) $@; \
 $(CXXLINK) -out:$$dllname -DLL $$extralinkoption \
 $$defflag -IMPLIB:$@ $(IMPORT_LIBRARY_FLAGS) \
 $^ $$extralibs $$resname; \
-$(MANIFESTTOOL) /outputresource:"$$dllname;#2" /manifest $$dllname.manifest; \
+$(MANIFESTTOOL) -outputresource:"$$dllname;#2" -manifest $$dllname.manifest; \
 $(CP) $@ $$slibname;
 endef
 
 # Export SharedLibrary
