diff --git a/configure.ac b/configure.ac
index de7e3eaa3..286d45d53 100644
--- a/configure.ac
+++ b/configure.ac
@@ -271,46 +271,11 @@ AC_CONFIG_COMMANDS([acconfig.h],
 AC_CONFIG_FILES(mk/beforeauto.mk mk/afterauto.mk GNUmakefile)
-AC_CONFIG_FILES(src/tool/omniidl/python2/scripts/omniidl)
 AC_CONFIG_FILES(src/tool/omniidl/python3/scripts/omniidl)
-AC_CONFIG_FILES(src/appl/GNUmakefile
-                src/appl/omniMapper/GNUmakefile
-                src/appl/omniNames/GNUmakefile
-                src/appl/utils/catior/GNUmakefile
-                src/appl/utils/convertior/GNUmakefile
-                src/appl/utils/GNUmakefile
-                src/appl/utils/genior/GNUmakefile
-                src/appl/utils/nameclt/GNUmakefile
-                src/GNUmakefile
-                src/examples/ami/GNUmakefile
-                src/examples/anyExample/GNUmakefile
-                src/examples/bidir/GNUmakefile
-                src/examples/boa/GNUmakefile
-                src/examples/call_back/GNUmakefile
-                src/examples/dii/GNUmakefile
-                src/examples/GNUmakefile
-                src/examples/dsi/GNUmakefile
-                src/examples/echo/GNUmakefile
-                src/examples/poa/GNUmakefile
-                src/examples/poa/implicit_activation/GNUmakefile
-                src/examples/poa/persistent_objref/GNUmakefile
-                src/examples/poa/servant_manager/GNUmakefile
-                src/examples/poa/threading/GNUmakefile
-                src/examples/poa/shortcut/GNUmakefile
-                src/examples/ssl_echo/GNUmakefile
-                src/examples/thread/GNUmakefile
-                src/examples/valuetype/GNUmakefile
-                src/examples/valuetype/simple/GNUmakefile
-                src/examples/ziop/GNUmakefile
+AC_CONFIG_FILES(src/GNUmakefile
                 src/lib/GNUmakefile
                 src/lib/omniORB/codesets/GNUmakefile
                 src/lib/omniORB/GNUmakefile
                 src/lib/omniORB/dynamic/GNUmakefile
-                src/lib/omniORB/python2/omniidl_be/cxx/GNUmakefile
-                src/lib/omniORB/python2/omniidl_be/cxx/dynskel/GNUmakefile
-                src/lib/omniORB/python2/omniidl_be/cxx/header/GNUmakefile
-                src/lib/omniORB/python2/omniidl_be/cxx/impl/GNUmakefile
-                src/lib/omniORB/python2/omniidl_be/cxx/skel/GNUmakefile
-                src/lib/omniORB/python2/omniidl_be/GNUmakefile
-                src/lib/omniORB/python2/GNUmakefile
                 src/lib/omniORB/python3/omniidl_be/cxx/GNUmakefile
                 src/lib/omniORB/python3/omniidl_be/cxx/dynskel/GNUmakefile
                 src/lib/omniORB/python3/omniidl_be/cxx/header/GNUmakefile
@@ -336,10 +301,6 @@ AC_CONFIG_FILES(src/appl/GNUmakefile
                 src/tool/omniidl/cxx/cccp/GNUmakefile
                 src/tool/omniidl/cxx/GNUmakefile
                 src/tool/omniidl/GNUmakefile
-                src/tool/omniidl/python2/GNUmakefile
-                src/tool/omniidl/python2/omniidl_be/GNUmakefile
-                src/tool/omniidl/python2/omniidl/GNUmakefile
-                src/tool/omniidl/python2/scripts/GNUmakefile
                 src/tool/omniidl/python3/GNUmakefile
                 src/tool/omniidl/python3/omniidl_be/GNUmakefile
                 src/tool/omniidl/python3/omniidl/GNUmakefile
diff --git a/mk/beforeauto.mk.in b/mk/beforeauto.mk.in
index 3c471a203..451f0255c 100644
--- a/mk/beforeauto.mk.in
+++ b/mk/beforeauto.mk.in
@@ -785,7 +785,7 @@ OPEN_SSL_CPPFLAGS = @OPEN_SSL_CPPFLAGS@
 #
 
 ifeq (@ENABLE_STATIC@,no)
-NoStaticLibrary = 1
+# NoStaticLibrary = 1
 endif
 
 #
diff --git a/mk/mklib.mk b/mk/mklib.mk
index fb83cf391..9b28a314b 100644
--- a/mk/mklib.mk
+++ b/mk/mklib.mk
@@ -100,6 +100,8 @@ $(shlib): $(patsubst %, shared/%, $(LIB_OBJS) $(LIB_SHARED_ONLY_OBJS))
 	@(namespec="$(namespec)" extralibs="$(imps) $(extralibs)"; \
          $(MakeCXXSharedLibrary))
 
+$(shlib): | $(staticlib)
+
 export:: $(shlib)
 	@(namespec="$(namespec)"; \
           $(ExportSharedLibrary))
@@ -209,6 +211,8 @@ $(dbugshlib): $(patsubst %, shareddebug/%, $(LIB_OBJS) $(LIB_SHARED_ONLY_OBJS))
 	(namespec="$(namespec)" debug=1 extralibs="$(dbugimps) $(extralibs)"; \
          $(MakeCXXSharedLibrary))
 
+$(dbugshlib): | $(dbuglib)
+
 export:: $(dbugshlib)
 	@(namespec="$(namespec)" debug=1; \
           $(ExportSharedLibrary))
diff --git a/mk/win32.mk b/mk/win32.mk
index 783402530..4b8d9f79b 100644
--- a/mk/win32.mk
+++ b/mk/win32.mk
@@ -172,7 +172,7 @@ endif
 # Stuff to generate executable binaries.
 #
 
-IMPORT_LIBRARY_FLAGS = $(patsubst %,$(LibPathPattern),$(IMPORT_LIBRARY_DIRS))
+IMPORT_LIBRARY_FLAGS = $(warning DIRS:$(IMPORT_LIBRARY_DIRS))$(patsubst %,$(LibPathPattern),$(IMPORT_LIBRARY_DIRS))
 
 define CXXExecutable
 (set -x; \
