diff --git a/mk/win32.mk b/mk/win32.mk
index 83f1b1687..741835b4b 100644
--- a/mk/win32.mk
+++ b/mk/win32.mk
@@ -43,10 +43,10 @@ endif
 #AR = $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/libwrapper $(XLN)
 #CXX = $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/clwrapper $(XLN)
 #CXXLINK	= $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/linkwrapper $(XLN)
-CXXMAKEDEPEND = $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/omkdepend $(MKDEPOPT) -D__cplusplus -D_MSC_VER
+CXXMAKEDEPEND = $(OMKDEPEND) -D__cplusplus -D_MSC_VER
 #CC = $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/clwrapper $(XLN)
 #CLINK = $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/linkwrapper $(XLN)
-CMAKEDEPEND = $(BASE_OMNI_TREE)/$(WRAPPER_FPATH)/omkdepend $(MKDEPOPT) -D_MSC_VER
+CMAKEDEPEND = $(OMKDEPEND) $(MKDEPOPT) -D_MSC_VER
 
 RCTOOL          = rc.exe
 MANIFESTTOOL    = true
@@ -145,7 +145,7 @@ BinPattern = %.exe
 define StaticLinkLibrary
 (set -x; \
  $(RM) $@; \
- $(AR) $@ $^; \
+ $(AR) -OUT:$@ $^; \
 )
 endef
 
@@ -439,12 +444,11 @@ endef
 # The pattern rules below ensured that the right compiler flags are used
 # to compile the source for the library.
 
+ifndef NoReleaseBuild
+
 static/%.o: %.cc
 	$(CXX) -c $(CXXDEBUGFLAGS) $(MSVC_STATICLIB_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdstatic\\$(LIB_NAME)$(major).pdb $<
 
-debug/%.o: %.cc
-	$(CXX) -c  $(MSVC_STATICLIB_CXXDEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fddebug\\$(LIB_NAME)$(major)d.pdb $<
-
 shared/%DynSK.o: %DynSK.cc
 	$(CXX) -c $(CXXDEBUGFLAGS) -DUSE_core_stub_in_nt_dll $(MSVC_DLL_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdshared\\ $<
 
@@ -454,6 +458,18 @@ shared/%SK.o: %SK.cc
 shared/%.o: %.cc
 	$(CXX) -c $(CXXDEBUGFLAGS) $(MSVC_DLL_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdshared\\ $<
 
+static/%.o: %.c
+	$(CC) -c $(CDEBUGFLAGS) $(MSVC_STATICLIB_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdstatic\\$(LIB_NAME)$(major).pdb $<
+
+shared/%.o: %.c
+	$(CC) -c $(CDEBUGFLAGS) $(MSVC_DLL_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ $<
+
+endif
+
+ifndef NoDebugBuild
+
+debug/%.o: %.cc
+	$(CXX) -c  $(MSVC_STATICLIB_CXXDEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fddebug\\$(LIB_NAME)$(major)d.pdb $<
 
 shareddebug/%DynSK.o: %DynSK.cc
 	$(CXX) -c  -DUSE_core_stub_in_nt_dll $(MSVC_DLL_CXXDEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdshareddebug\\ $<
@@ -464,19 +480,14 @@ shareddebug/%SK.o: %SK.cc
 shareddebug/%.o: %.cc
 	$(CXX) -c  $(MSVC_DLL_CXXDEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdshareddebug\\ $<
 
-static/%.o: %.c
-	$(CC) -c $(CDEBUGFLAGS) $(MSVC_STATICLIB_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdstatic\\$(LIB_NAME)$(major).pdb $<
-
 debug/%.o: %.c
 	$(CC) -c $(MSVC_STATICLIB_CXXDEBUGFLAGS) $(CPPFLAGS) -Fo$@ -Fdstatic\\$(LIB_NAME)$(major)d.pdb $<
 
-shared/%.o: %.c
-	$(CC) -c $(CDEBUGFLAGS) $(MSVC_DLL_CXXNODEBUGFLAGS) $(CPPFLAGS) -Fo$@ $<
-
-
 shareddebug/%.o: %.c
 	$(CC) -c  $(MSVC_DLL_CXXDEBUGFLAGS) $(CPPFLAGS) -Fo$@ $<
 
+endif
+
 
 #
 # Replacements for implicit rules
diff --git a/src/lib/omniORB/orbcore/dir.mk b/src/lib/omniORB/orbcore/dir.mk
index 634e6b1e2..4370fed07 100644
--- a/src/lib/omniORB/orbcore/dir.mk
+++ b/src/lib/omniORB/orbcore/dir.mk
@@ -220,6 +220,7 @@ include $(BASE_OMNI_TREE)/mk/mklib.mk
 
 #########################################################################
 ifdef Win32Platform
+ifndef NoReleaseBuild
 
 stublib = static/$(patsubst %,$(LibNoDebugPattern),msvcstub)
 
@@ -236,6 +237,8 @@ clean::
 
 veryclean::
 	$(RM) $(stublib)
+endif
+ifndef NoDebugBuild
 
 stubdblib = debug/$(patsubst %,$(LibDebugPattern),msvcstub)
 
@@ -254,6 +257,7 @@ veryclean::
 	$(RM) $(stubdblib)
 
 endif
+endif
 
 #########################################################################
 ifdef OPEN_SSL_ROOT
diff --git a/src/services/mklib/dynstublib/dir.mk b/src/services/mklib/dynstublib/dir.mk
index e128aa87a..f2e1448b3 100644
--- a/src/services/mklib/dynstublib/dir.mk
+++ b/src/services/mklib/dynstublib/dir.mk
@@ -17,6 +17,14 @@ ifdef Win32Platform
   MSVC_STATICLIB_CXXDEBUGFLAGS += -D_WINSTATIC
 endif
 
+ifdef Win32Platform
+# in case of Win32 lossage:
+dynimps := COS$(OMNIORB_MAJOR_VERSION)$(OMNIORB_MINOR_VERSION)$(OMNIORB_MICRO_VERSION)_rt.lib \
+           $(patsubst $(DLLDebugSearchPattern),$(DLLNoDebugSearchPattern), $(OMNIORB_LIB))
+else
+dynimps := $(OMNIORB_COS_LIB) $(OMNIORB_LIB)
+endif
+
 ifndef NoReleaseBuild
 all:: mkstatic mkshared
 
@@ -75,14 +83,6 @@ dynskshared   = shared/$(shell $(SharedLibraryFullName) $(dynsknamespec))
 
 MDFLAGS += -p shared/
 
-ifdef Win32Platform
-# in case of Win32 lossage:
-dynimps := COS$(OMNIORB_MAJOR_VERSION)$(OMNIORB_MINOR_VERSION)$(OMNIORB_MICRO_VERSION)_rt.lib \
-           $(patsubst $(DLLDebugSearchPattern),$(DLLNoDebugSearchPattern), $(OMNIORB_LIB))
-else
-dynimps := $(OMNIORB_COS_LIB) $(OMNIORB_LIB)
-endif
-
 mkshared::
 	@(dir=shared; $(CreateDir))
 
diff --git a/src/services/mklib/stublib/dir.mk b/src/services/mklib/stublib/dir.mk
index 87f7ec22b..ce46f6c5f 100644
--- a/src/services/mklib/stublib/dir.mk
+++ b/src/services/mklib/stublib/dir.mk
@@ -115,7 +115,7 @@ endif
 ##############################################################################
 # Build debug libraries for Win32
 ##############################################################################
-ifdef NoDebugBuild
+ifndef NoDebugBuild
 ifdef Win32Platform
 
 all:: mkstaticdbug mkshareddbug
