diff --git a/acinclude.m4 b/acinclude.m4
index cef87f8bd..cf61556bd 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -433,8 +433,8 @@ AC_SUBST(ENABLE_LONGDOUBLE, $omni_cv_enable_longdouble)
 
 dnl Enable ZIOP
 AC_DEFUN([OMNI_ENABLE_ZIOP],
-[AC_CHECK_LIB(z,compressBound,omni_cv_enable_ziop_zlib=yes,omni_cv_enable_ziop_zlib=no)
-AC_CHECK_LIB(zstd,ZSTD_compress,omni_cv_enable_ziop_zstd=yes,omni_cv_enable_ziop_zstd=no)
+[AC_SEARCH_LIBS(compressBound,[z zlib zlibd],omni_cv_enable_ziop_zlib=yes,omni_cv_enable_ziop_zlib=no)
+AC_SEARCH_LIBS(ZSTD_compress,[zstd zstdd],omni_cv_enable_ziop_zstd=yes,omni_cv_enable_ziop_zstd=no)
 omni_cv_enable_ziop="no"
 if test "$omni_cv_enable_ziop_zlib" = "yes"; then
   omni_cv_enable_ziop="yes"
diff --git a/mk/win32.mk b/mk/win32.mk
index 741835b4b..7a5bff934 100644
--- a/mk/win32.mk
+++ b/mk/win32.mk
@@ -580,8 +580,7 @@ OMNIORB_IDL = $(OMNIORB_IDL_ONLY) $(OMNIORB_IDL_ANY_FLAGS)
 OMNIORB_CPPFLAGS = -D__OMNIORB4__ -I$(CORBA_STUB_DIR) $(OMNITHREAD_CPPFLAGS)
 OMNIORB_IDL_OUTPUTDIR_PATTERN = -C%
 
-msvc_work_around_stub = $(patsubst %,$(LibPattern),msvcstub) \
-                        -NODEFAULTLIB:libcmt.lib -NODEFAULTLIB:libcmtd.lib 
+msvc_work_around_stub = $(patsubst %,$(LibPattern),msvcstub) 
 
 OMNIORB_LIB = $(omniorb_dll_name) \
 		$(omnidynamic_dll_name) \
diff --git a/src/tool/omniidl/cxx/dir.mk b/src/tool/omniidl/cxx/dir.mk
index 3b5a60b87..31cbc33e7 100644
--- a/src/tool/omniidl/cxx/dir.mk
+++ b/src/tool/omniidl/cxx/dir.mk
@@ -196,7 +196,7 @@ ifdef MinGW32Build
 PYLIB     := -lpython$(subst .,,$(PYVERSION))
 CXXLINKOPTIONS += $(patsubst %,-L%,$(PYLIBDIR))
 else
-PYLIB     = python$(subst .,,$(PYVERSION)).lib
+PYLIB     = python$(subst .,,$(PYVERSION)).lib Advapi32.lib
 CXXLINKOPTIONS += $(patsubst %,-libpath:%,$(PYLIBDIR))
 endif
 
