diff --git a/cmake/jkqtplotter_deployqt.cmake b/cmake/jkqtplotter_deployqt.cmake
index a5a0da5..305e176 100644
--- a/cmake/jkqtplotter_deployqt.cmake	
+++ b/cmake/jkqtplotter_deployqt.cmake
@@ -1,6 +1,6 @@
 function(jkqtplotter_deployqt TARGET_NAME)
     #use windeploy
-    if (WIN32)
+    if (0)
         get_target_property(_qmake_executable Qt${QT_VERSION_MAJOR}::qmake IMPORTED_LOCATION)
         get_filename_component(_qt_bin_dir "${_qmake_executable}" DIRECTORY)
         find_program(WINDEPLOYQT_ENV_SETUP qtenv2.bat HINTS "${_qt_bin_dir}")
@@ -23,5 +23,5 @@ function(jkqtplotter_deployqt TARGET_NAME)
             COMMAND "${WINDEPLOYQT_ENV_SETUP}" && "${WINDEPLOYQT_EXECUTABLE}" \"$<TARGET_FILE:${TARGET_NAME}>\"
             COMMENT "Running windeployqt ... "
         )
-    endif(WIN32)
+    endif()
 endfunction(jkqtplotter_deployqt)
