diff --git a/cmake/jkqtplotter_common_qtsettings.cmake b/cmake/jkqtplotter_common_qtsettings.cmake
index 85a39ae..6085291 100644
--- a/cmake/jkqtplotter_common_qtsettings.cmake	
+++ b/cmake/jkqtplotter_common_qtsettings.cmake
@@ -10,11 +10,3 @@ if(${QT_VERSION_MAJOR} VERSION_GREATER_EQUAL "6")
     find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS OpenGLWidgets)
 endif()
 
-# configure compiler
-if(${QT_VERSION_MAJOR} VERSION_LESS "6" )
-    set(CMAKE_CXX_STANDARD 11)
-    set(CMAKE_CXX_STANDARD_REQUIRED on)
-else()
-    set(CMAKE_CXX_STANDARD 17)
-    set(CMAKE_CXX_STANDARD_REQUIRED on)
-endif()
diff --git a/lib/jkqtcommon/CMakeLists.txt b/lib/jkqtcommon/CMakeLists.txt
index 40f52c5..db5ee3a 100644
--- a/lib/jkqtcommon/CMakeLists.txt	
+++ b/lib/jkqtcommon/CMakeLists.txt
@@ -132,12 +132,8 @@ if(JKQtPlotter_BUILD_STATIC_LIBS)
     set_property(TARGET ${lib_name} PROPERTY VERSION "${PROJECT_VERSION}")
     set_property(TARGET ${lib_name} PROPERTY OUTPUT_NAME  "${lib_name_decorated}")
     if(${QT_VERSION_MAJOR} VERSION_LESS "6")
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 11)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_11)
     else()
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 17)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_17)
     endif()
     if(MINGW)
diff --git a/lib/jkqtfastplotter/CMakeLists.txt b/lib/jkqtfastplotter/CMakeLists.txt
index 17878ed..250f3a1 100644
--- a/lib/jkqtfastplotter/CMakeLists.txt	
+++ b/lib/jkqtfastplotter/CMakeLists.txt
@@ -84,13 +84,9 @@ if(JKQtPlotter_BUILD_STATIC_LIBS)
     endif()
     target_link_libraries(${lib_name} PUBLIC Qt${QT_VERSION_MAJOR}::Core Qt${QT_VERSION_MAJOR}::Gui Qt${QT_VERSION_MAJOR}::Widgets Qt${QT_VERSION_MAJOR}::PrintSupport Qt${QT_VERSION_MAJOR}::OpenGL JKQTCommonLib)
     if(${QT_VERSION_MAJOR} VERSION_LESS "6")
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 11)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_11)
     else()
         target_link_libraries(${lib_name} PUBLIC Qt${QT_VERSION_MAJOR}::OpenGLWidgets)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 17)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_17)
     endif()
     target_include_directories(${lib_name} PUBLIC
diff --git a/lib/jkqtmathtext/CMakeLists.txt b/lib/jkqtmathtext/CMakeLists.txt
index b119d10..991f935 100644
--- a/lib/jkqtmathtext/CMakeLists.txt	
+++ b/lib/jkqtmathtext/CMakeLists.txt
@@ -82,12 +82,8 @@ if(JKQtPlotter_BUILD_STATIC_LIBS)
     set_property(TARGET ${lib_name} PROPERTY VERSION "${PROJECT_VERSION}")
     set_property(TARGET ${lib_name} PROPERTY OUTPUT_NAME  "${lib_name_decorated}")
     if(${QT_VERSION_MAJOR} VERSION_LESS "6")
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 11)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_11)
     else()
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 17)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_17)
     endif()
     if(MINGW)
diff --git a/lib/jkqtplotter/CMakeLists.txt b/lib/jkqtplotter/CMakeLists.txt
index 8a85a0e..24d52e0 100644
--- a/lib/jkqtplotter/CMakeLists.txt	
+++ b/lib/jkqtplotter/CMakeLists.txt
@@ -236,12 +236,8 @@ if(JKQtPlotter_BUILD_STATIC_LIBS)
     set_property(TARGET ${lib_name} PROPERTY VERSION "${PROJECT_VERSION}")
     set_property(TARGET ${lib_name} PROPERTY OUTPUT_NAME  "${lib_name_decorated}")
     if(${QT_VERSION_MAJOR} VERSION_LESS "6")
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 11)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_11)
     else()
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD 17)
-        set_property(TARGET ${lib_name} PROPERTY CXX_STANDARD_REQUIRED TRUE)
         target_compile_features(${lib_name} PUBLIC cxx_std_17)
     endif()
     if(MINGW)
