diff --git a/c7df1582d3-3df89c79f1.clean/cli/Makefile b/c7df1582d3-3df89c79f1/cli/Makefile
index 60fd97c3c..bfa9994a1 100644
--- a/c7df1582d3-3df89c79f1.clean/cli/Makefile
+++ b/c7df1582d3-3df89c79f1/cli/Makefile
@@ -111,7 +111,7 @@ STRIP_EXPORTED_FUNCS := $(shell $(CPP_STDOUT) -I$(JULIAHOME)/src $(SRCDIR)/list_
 endif
 
 $(build_shlibdir)/libjulia.$(JL_MAJOR_MINOR_SHLIB_EXT): $(LIB_OBJS) $(SRCDIR)/list_strip_symbols.h | $(build_shlibdir) $(build_libdir)
-	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@.tmp) $(LOADER_CFLAGS) -DLIBRARY_EXPORTS -shared $(SHIPFLAGS) $(LIB_OBJS) -o $@ \
+	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@.tmp) $(LOADER_CFLAGS) -DJULIA_LIBRARY_EXPORTS -shared $(SHIPFLAGS) $(LIB_OBJS) -o $@ \
 		$(JLIBLDFLAGS) $(call SONAME_FLAGS,libjulia.$(JL_MAJOR_SHLIB_EXT)))
 	@$(INSTALL_NAME_CMD)libjulia.$(JL_MAJOR_SHLIB_EXT) $@
 ifeq ($(OS), WINNT)
@@ -122,7 +122,7 @@ ifeq ($(OS), WINNT)
 endif
 
 $(build_shlibdir)/libjulia-debug.$(JL_MAJOR_MINOR_SHLIB_EXT): $(LIB_DOBJS) $(SRCDIR)/list_strip_symbols.h | $(build_shlibdir) $(build_libdir)
-	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@.tmp) $(LOADER_CFLAGS) -DLIBRARY_EXPORTS -shared $(DEBUGFLAGS) $(LIB_DOBJS) -o $@ \
+	@$(call PRINT_LINK, $(CC) $(call IMPLIB_FLAGS,$@.tmp) $(LOADER_CFLAGS) -DJULIA_LIBRARY_EXPORTS -shared $(DEBUGFLAGS) $(LIB_DOBJS) -o $@ \
 		$(JLIBLDFLAGS) $(call SONAME_FLAGS,libjulia-debug.$(JL_MAJOR_SHLIB_EXT)))
 	@$(INSTALL_NAME_CMD)libjulia-debug.$(JL_MAJOR_SHLIB_EXT) $@
 ifeq ($(OS), WINNT)
diff --git a/c7df1582d3-3df89c79f1.clean/cli/jl_exports.h b/c7df1582d3-3df89c79f1/cli/jl_exports.h
index 05e7d6b77..06cc31df2 100644
--- a/c7df1582d3-3df89c79f1.clean/cli/jl_exports.h
+++ b/c7df1582d3-3df89c79f1/cli/jl_exports.h
@@ -7,6 +7,22 @@
 #include "../src/jl_exported_data.inc"
 #include "../src/jl_exported_funcs.inc"
 
+#ifdef _OS_WINDOWS_
+# ifdef JULIA_LIBRARY_EXPORTS
+#  define JL_DLLEXPORT __declspec(dllexport)
+# else
+#  define JL_DLLEXPORT __declspec(dllimport)
+# endif
+#define JL_HIDDEN
+#else
+# if defined(JULIA_LIBRARY_EXPORTS) && defined(_OS_LINUX_)
+#  define JL_DLLEXPORT __attribute__ ((visibility("protected")))
+# else
+#  define JL_DLLEXPORT __attribute__ ((visibility("default")))
+# endif
+#define JL_HIDDEN    __attribute__ ((visibility("hidden")))
+#endif
+
 // Define pointer data as `const void * $(name);`
 #define XX(name)    JL_DLLEXPORT const void * name;
 JL_EXPORTED_DATA_POINTERS(XX)
@@ -74,3 +90,19 @@ static anonfunc **const jl_codegen_exported_func_addrs[] = {
     NULL
 };
 #undef XX
+
+#ifdef _OS_WINDOWS_
+# ifdef LIBRARY_EXPORTS
+#  define JL_DLLEXPORT __declspec(dllexport)
+# else
+#  define JL_DLLEXPORT __declspec(dllimport)
+# endif
+#define JL_HIDDEN
+#else
+# if defined(LIBRARY_EXPORTS) && defined(_OS_LINUX_)
+#  define JL_DLLEXPORT __attribute__ ((visibility("protected")))
+# else
+#  define JL_DLLEXPORT __attribute__ ((visibility("default")))
+# endif
+#define JL_HIDDEN    __attribute__ ((visibility("hidden")))
+#endif
diff --git a/c7df1582d3-3df89c79f1.clean/src/Makefile b/c7df1582d3-3df89c79f1/src/Makefile
index 22ace44b2..d5922da4b 100644
--- a/c7df1582d3-3df89c79f1.clean/src/Makefile
+++ b/c7df1582d3-3df89c79f1/src/Makefile
@@ -157,7 +157,7 @@ LIBJULIA_PATH_REL := libjulia
 endif
 
 COMMON_LIBPATHS := -L$(build_libdir) -L$(build_shlibdir)
-RT_LIBS := $(WHOLE_ARCHIVE) $(LIBUV) $(WHOLE_ARCHIVE) $(LIBUTF8PROC) $(NO_WHOLE_ARCHIVE) $(LIBUNWIND) $(RT_LLVMLINK) $(OSLIBS)
+RT_LIBS := $(LIBUV) $(LIBUTF8PROC) $(NO_WHOLE_ARCHIVE) $(LIBUNWIND) $(RT_LLVMLINK) $(OSLIBS)
 CG_LIBS := $(LIBUNWIND) $(CG_LLVMLINK) $(OSLIBS)
 RT_DEBUG_LIBS := $(COMMON_LIBPATHS) $(WHOLE_ARCHIVE) $(BUILDDIR)/flisp/libflisp-debug.a $(WHOLE_ARCHIVE) $(BUILDDIR)/support/libsupport-debug.a libjulia-debug.dll.a $(RT_LIBS)
 CG_DEBUG_LIBS := $(COMMON_LIBPATHS) $(CG_LIBS) libjulia-debug.dll.a libjulia-internal-debug.dll.a
@@ -364,13 +364,13 @@ $(BUILDDIR)/julia_version.h: $(JULIAHOME)/VERSION
 CXXLD = $(CXX) -Xlinker -DLL
 
 $(build_shlibdir)/libjulia-internal.$(JL_MAJOR_MINOR_SHLIB_EXT): $(SRCDIR)/julia.expmap $(OBJS) $(BUILDDIR)/flisp/libflisp.a $(BUILDDIR)/support/libsupport.a
-	@$(call PRINT_LINK, $(CXXLD) $(call IMPLIB_FLAGS,$@) $(JCXXFLAGS) $(JL_CXXFLAGS) $(CXXLDFLAGS) $(SHIPFLAGS) $(OBJS) $(RPATH_LIB) -o $@ \
+	@$(call PRINT_LINK, $(CXXLD) $(call IMPLIB_FLAGS,$@) $(JCXXFLAGS) $(JL_CXXFLAGS) $(CXXLDFLAGS) $(SHIPFLAGS) $(OBJS) $(RPATH_LIB) $(WHOLE_ARCHIVE) -llibjulia.dll -o $@ \
 		$(JLDFLAGS) $(JLIBLDFLAGS) $(RT_RELEASE_LIBS) $(call SONAME_FLAGS,libjulia-internal.$(JL_MAJOR_SHLIB_EXT)))
 	@$(INSTALL_NAME_CMD)libjulia-internal.$(SHLIB_EXT) $@
 	$(DSYMUTIL) $@
 
 $(build_shlibdir)/libjulia-internal-debug.$(JL_MAJOR_MINOR_SHLIB_EXT): $(SRCDIR)/julia.expmap $(DOBJS) $(BUILDDIR)/flisp/libflisp-debug.a $(BUILDDIR)/support/libsupport-debug.a
-	@$(call PRINT_LINK, $(CXXLD) $(call IMPLIB_FLAGS,$@) $(JCXXFLAGS) $(JL_CXXFLAGS) $(CXXLDFLAGS) $(DEBUGFLAGS) $(DOBJS) $(RPATH_LIB) -o $@ \
+	@$(call PRINT_LINK, $(CXXLD) $(call IMPLIB_FLAGS,$@) $(JCXXFLAGS) $(JL_CXXFLAGS) $(CXXLDFLAGS) $(DEBUGFLAGS) $(DOBJS) $(RPATH_LIB) $(WHOLE_ARCHIVE) -llibjulia.dll -o $@ \
 		$(JLDFLAGS) $(JLIBLDFLAGS) $(RT_DEBUG_LIBS) $(call SONAME_FLAGS,libjulia-internal-debug.$(JL_MAJOR_SHLIB_EXT)))
 	@$(INSTALL_NAME_CMD)libjulia-internal-debug.$(SHLIB_EXT) $@
 	$(DSYMUTIL) $@
diff --git a/c7df1582d3-3df89c79f1.clean/src/julia.h b/c7df1582d3-3df89c79f1/src/julia.h
index f5d7c6e29..6b252180a 100644
--- a/c7df1582d3-3df89c79f1.clean/src/julia.h
+++ b/c7df1582d3-3df89c79f1/src/julia.h
@@ -3,7 +3,7 @@
 #ifndef JULIA_H
 #define JULIA_H
 
-#ifdef LIBRARY_EXPORTS
+#ifdef JULIA_LIBRARY_EXPORTS
 // Generated file, needs to be searched in include paths so that the builddir
 // retains priority
 #include <jl_internal_funcs.inc>
@@ -70,7 +70,7 @@
 typedef struct _jl_taggedvalue_t jl_taggedvalue_t;
 typedef struct _jl_tls_states_t *jl_ptls_t;
 
-#ifdef LIBRARY_EXPORTS
+#ifdef JULIA_LIBRARY_EXPORTS
 #include "uv.h"
 #endif
 #include "julia_atomics.h"
@@ -1989,7 +1989,7 @@ void (jl_longjmp)(jmp_buf _Buf, int _Value);
 JL_DLLEXPORT int (ijl_setjmp)(jmp_buf _Buf);
 void (ijl_longjmp)(jmp_buf _Buf, int _Value);
 #endif
-#ifdef LIBRARY_EXPORTS
+#ifdef JULIA_LIBRARY_EXPORTS
 #define jl_setjmp_f ijl_setjmp
 #define jl_setjmp_name "ijl_setjmp"
 #define jl_setjmp(a,b) ijl_setjmp(a)
diff --git a/c7df1582d3-3df89c79f1.clean/src/threading.c b/c7df1582d3-3df89c79f1/src/threading.c
index 0c159718b..75a190dad 100644
--- a/c7df1582d3-3df89c79f1.clean/src/threading.c
+++ b/c7df1582d3-3df89c79f1/src/threading.c
@@ -643,7 +643,7 @@ void jl_start_threads(void)
         memset(mask, 0, cpumasksize);
         mask[0] = 1;
         uvtid = uv_thread_self();
-        uv_thread_setaffinity(&uvtid, mask, NULL, cpumasksize);
+        //uv_thread_setaffinity(&uvtid, mask, NULL, cpumasksize);
         mask[0] = 0;
     }
 
@@ -657,10 +657,10 @@ void jl_start_threads(void)
         uv_thread_create(&uvtid, jl_threadfun, t);
         if (exclusive) {
             mask[i] = 1;
-            uv_thread_setaffinity(&uvtid, mask, NULL, cpumasksize);
+            //uv_thread_setaffinity(&uvtid, mask, NULL, cpumasksize);
             mask[i] = 0;
         }
-        uv_thread_detach(&uvtid);
+        //uv_thread_detach(&uvtid);
     }
 
     uv_barrier_wait(&thread_init_done);
