diff --git a/Filters/General/vtkTableFFT.cxx b/Filters/General/vtkTableFFT.cxx
index db2e744a5db0a956a617db39a8d9df24f5a7f368..79061f047722e3d1b4c539f6c4fe3a6077b44ca4 100644
--- a/Filters/General/vtkTableFFT.cxx
+++ b/Filters/General/vtkTableFFT.cxx
@@ -55,7 +55,7 @@ typename std::iterator_traits<InputIt>::value_type WindowEnergy(InputIt begin, I
 
 // Easy access to the right windowing function using vtkTableFFT enumeration.
 // clang-format off
-constexpr static std::array<vtkFFT::WindowGenerator, vtkTableFFT::MAX_WINDOWING_FUNCTION> WindowingFunctionsList =
+static const std::array<vtkFFT::WindowGenerator, vtkTableFFT::MAX_WINDOWING_FUNCTION> WindowingFunctionsList =
 {
   vtkFFT::HanningGenerator,
   vtkFFT::BartlettGenerator,
