diff --git a/CMake/vtkInstallCMakePackage.cmake b/CMake/vtkInstallCMakePackage.cmake
index be1da92..562a7c5 100644
--- a/CMake/vtkInstallCMakePackage.cmake
+++ b/CMake/vtkInstallCMakePackage.cmake
@@ -80,7 +80,6 @@ configure_file(
 
 set(vtk_cmake_module_files
   Finddouble-conversion.cmake
-  FindEigen3.cmake
   FindEXPAT.cmake
   FindExprTk.cmake
   FindFFMPEG.cmake
diff --git a/ThirdParty/eigen/CMakeLists.txt b/ThirdParty/eigen/CMakeLists.txt
index 1edabc4..aa922dc 100644
--- a/ThirdParty/eigen/CMakeLists.txt
+++ b/ThirdParty/eigen/CMakeLists.txt
@@ -6,7 +6,7 @@ vtk_module_third_party(
     STANDARD_INCLUDE_DIRS
   EXTERNAL
     PACKAGE Eigen3
-    TARGETS Eigen3::Eigen3
+    TARGETS Eigen3::Eigen
     STANDARD_INCLUDE_DIRS)
 
 configure_file(
