diff --git a/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprLex.cxx b/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprLex.cxx
index 24a3518..9070e0c 100644
--- a/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprLex.cxx	
+++ b/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprLex.cxx
@@ -702,6 +702,8 @@ extern int dice_yylex (void);
 #define YY_RULE_SETUP \
   YY_USER_ACTION
 
+#define register
+
 /** The main scanner function which does all the work.
  */
 YY_DECL
diff --git a/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprYacc.cxx b/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprYacc.cxx
index fca52f7..7fd3560 100644
--- a/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprYacc.cxx	
+++ b/ThirdParty/xdmf2/vtkxdmf2/libsrc/XdmfExprYacc.cxx
@@ -260,6 +260,8 @@ union yyalloc
      ((N) * (sizeof (short) + sizeof (YYSTYPE))                                \
       + YYSTACK_GAP_MAXIMUM)
 
+#define register
+
 /* Copy COUNT objects from FROM to TO.  The source and destination do
    not overlap.  */
 # ifndef YYCOPY
