diff --git a/frame/base/bli_cpuid.c b/frame/base/bli_cpuid.c
index 8ea80cb14f..1db3137dda 100644
--- a/frame/base/bli_cpuid.c
+++ b/frame/base/bli_cpuid.c
@@ -60,6 +60,8 @@
 
 #if defined(__x86_64__) || defined(_M_X64) || defined(__i386) || defined(_M_IX86)
 
+#include "cpuid.h"
+
 arch_t bli_cpuid_query_id( void )
 {
 	uint32_t vendor, family, model, features;
diff --git a/frame/base/bli_cpuid.h b/frame/base/bli_cpuid.h
index b7ba96ef94..e06e62b4cf 100644
--- a/frame/base/bli_cpuid.h
+++ b/frame/base/bli_cpuid.h
@@ -125,8 +125,6 @@ static bool_t bli_cpuid_has_features( uint32_t have, uint32_t want )
 
 #if defined(__x86_64__) || defined(_M_X64) || defined(__i386) || defined(_M_IX86)
 
-#include "cpuid.h"
-
 void get_cpu_name( char *cpu_name );
 int  vpu_count( void );
 
