diff --git a/databases/readers/MFIXCDF/avtMFIXCDFFileFormat.C b/databases/readers/MFIXCDF/avtMFIXCDFFileFormat.C
index 12429f4..5e4c671 100644
--- a/databases/readers/MFIXCDF/avtMFIXCDFFileFormat.C	
+++ b/databases/readers/MFIXCDF/avtMFIXCDFFileFormat.C
@@ -13,6 +13,7 @@
 #include <math.h>
 #include <visitstream.h>
 #include <sstream>
+#include <memory>
 
 #include <vtkCellData.h>
 #include <vtkDataArray.h>
@@ -51,7 +52,7 @@
 #define cbrt(x) (pow(x, 1.0/3.0))
 #endif
 
-using std::auto_ptr;
+using std::unique_ptr;
 using std::vector;
 using std::ostringstream;
 
@@ -1584,13 +1585,13 @@ avtMFIXCDFFileFormat::GetVectorVar(int domain, const char *varname)
         nzvals= (widths[2]+3);
     }
 
-    // This trick with auto_ptr makes xvec, yvec and zvec get deleted
+    // This trick with unique_ptr makes xvec, yvec and zvec get deleted
     // when they go out of scope.
-    auto_ptr< vector<float> > xvec(new vector<float>(totZones));
+    unique_ptr< vector<float> > xvec(new vector<float>(totZones));
     float* xdata= &(*xvec)[0];
-    auto_ptr< vector<float> > yvec(new vector<float>(totZones));
+    unique_ptr< vector<float> > yvec(new vector<float>(totZones));
     float* ydata= &(*yvec)[0];
-    auto_ptr< vector<float> > zvec(new vector<float>(totZones));
+    unique_ptr< vector<float> > zvec(new vector<float>(totZones));
     float* zdata= &(*zvec)[0];
 
     if (!strncmp(varname,"Vel_",4))
